caf-substitutor-dropwizard/src/main/java/com/github/cafapi/util/dropwizard/CustomSubstitutor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 Open Text.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.util.dropwizard;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.text.StringSubstitutor;
+import org.apache.commons.text.lookup.StringLookup;
+import org.apache.commons.text.lookup.StringLookupFactory;
+
+import com.hpe.caf.secret.SecretUtil;
+
+/**
+ * A custom implementation of StringSubstitutor that provides additional functionality
+ * for variable substitution, including secret lookup.
+ * <p>
+ * This class extends StringSubstitutor and configures it with a custom StringLookup
+ * that combines secret lookup and environment variable lookup capabilities.
+ */
+public final class CustomSubstitutor extends StringSubstitutor
```

Comment:
Before I looked at this I thought it should be called something like `EnvironmentVariableAndSecretSubstitutor` but now that I've looked at how generic and extendable the implementation that you've written is I'm wondering if something like `CafConfigSubstitutor` would be more appropriate?